cmd/web-handlers.go

```diff
@@ -728,12 +728,23 @@ func (web *webAPIHandlers) ListAllBucketPolicies(r *http.Request, args *ListAllB
 	if !isHTTPRequestValid(r) {
 		return toJSONError(errAuthentication)
 	}
-
-	policyInfo, err := readBucketAccessPolicy(objectAPI, args.BucketName)
+	// FIXME: remove this code when S3 layer for gateway and server is unified.
+	var policyInfo policy.BucketAccessPolicy
+	var err error
+	if layer, ok := objectAPI.(*s3Objects); ok {
+		policyInfo, err = layer.GetBucketPolicies(args.BucketName)	
+	} else if layer, ok := objectAPI.(*azureObjects); ok {
+		policyInfo, err = layer.GetBucketPolicies(args.BucketName)
+	} else {
+		policyInfo, err = readBucketAccessPolicy(objectAPI, args.BucketName)
+		 
+	}
```

Comment:
@poornas you can use go's switch statement for this which is more idiomatic in this situation:
```
switch layer := objectAPI.(type) {
case *s3Objects:
    policyInfo, err = layer.GetBucketPolicies(args.BucketName)
case *azureObjects:
    policyInfo, err = layer.GetBucketPolicies(args.BucketName)
default:
    policyInfo, err = readBucketAccessPolicy(objectAPI, args.BucketName)
}
```
You can make similar changes at other places where this is needed